taskt/Core/Automation/Commands/Variable Commands/AddListItemCommand.cs

```diff
@@ -0,0 +1,108 @@
+using taskt.Core.Automation.Attributes.ClassAttributes;
+using taskt.Core.Automation.Attributes.PropertyAttributes;
+using taskt.Core.Automation.Engine;
+using taskt.Core.Utilities.CommonUtilities;
+using taskt.UI.CustomControls;
+using taskt.UI.Forms;
+
+namespace taskt.Core.Automation.Commands
+{
+    [Serializable]
+    [Group("Variable Commands")]
+    [Description("This command adds an item to an existing list variable.")]
+    public class AddListItemCommand : ScriptCommand
+    {
+        [XmlAttribute]
+        [PropertyDescription("List Item")]
+        [InputSpecification("Enter the list item for the variable.")]
+        [SampleUsage("Hello || {vText}")]
+        [Remarks("List item can only be a String, DataTable, MailItem or IWebElement.")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
+        public string v_ListItem { get; set; }
+
+        [XmlAttribute]
+        [PropertyDescription("Output List Variable")]
+        [InputSpecification("Select or provide a variable from the variable list.")]
+        [SampleUsage("vUserVariable")]
+        [Remarks("If you have enabled the setting **Create Missing Variables at Runtime** then you are not required" +
+                  " to pre-define your variables; however, it is highly recommended.")]
+        public string v_OutputUserVariableName { get; set; }
+
+        public AddListItemCommand()
+        {
+            CommandName = "AddListItemCommand";
+            SelectionName = "Add List Item";
+            CommandEnabled = true;
+            CustomRendering = true;
+        }
+
+        public override void RunCommand(object sender)
+        {
+            //get sending instance
+            var engine = (AutomationEngineInstance)sender;
+
+            var listVariable = VariableMethods.LookupVariable(engine, v_OutputUserVariableName);
+
+            if (listVariable != null)
+            {
+                if (listVariable.VariableValue is List<string>)
+                {
+                    ((List<string>)listVariable.VariableValue).Add(v_ListItem.ConvertToUserVariable(engine));
+                }
+                else if (listVariable.VariableValue is List<DataTable>)
+                {
+                    var dataTable = VariableMethods.LookupVariable(engine, v_ListItem).VariableValue;
+                    if (!(dataTable is DataTable))
+                        throw new System.Exception("Invalid List Item type, please provide valid List Item type.");
+                    ((List<DataTable>)listVariable.VariableValue).Add((DataTable)dataTable);
+                }
+                else if (listVariable.VariableValue is List<MailItem>)
+                {
+                    var mailItem = VariableMethods.LookupVariable(engine, v_ListItem).VariableValue;
+                    if (!(mailItem is MailItem))
+                        throw new System.Exception("Invalid List Item type, please provide valid List Item type."); 
```

Comment:
please remove namespaces

Reply:
I did it consciously as there is an ambiguity between `System.Exception` and `Microsoft.Office.Interop.Outlook.Exception`

Reply:
ok